src/qat/purr/integrations/qasm.py

```diff
@@ -944,16 +737,335 @@ def __repr__(self):
         return f"Partial instance of {type(self._ref_instance).__name__}"
 
 
+class Qasm3ParserBase(AbstractParser, QASMVisitor):
+
+    def parser_language(self) -> Languages:
+        return Languages.Qasm3
+
+    def _includes_standard_gates(self, qasm_str: str) -> bool:
```

Comment:
why do you we need to do this? Are the headers not in the parsed program?

Reply:
We can but it's a change in functionality. At present standard gates have been automatically added even without the includes. :thinking: